### PR TITLE
feat(organizations): scope assignments under organization group URLs

### LINKED ISSUE #7721: Feature: Scope assignments under organization-owned group URLs
### Is your feature request related to a problem? Please describe

Organization-owned groups can now be accessed through organization-scoped URLs **(/organizations/:org/groups/:id)**, but their assignments still only work through the flat, non-org-scoped routes (/groups/:group_id/assignments/...). This is inconsistent: when a mentor or admin is working inside an organization context and opens a group's assignments, the URLs drop the organization scope. It also means organization admins and mentors cannot manage assignments in classrooms created by other mentors, because assignment authorization only recognizes the group's primary mentor, not organization-level roles.

### Describe the solution you'd like

Make assignments work fully under organization-scoped group URLs, matching how the groups themselves already behave:

- Assignments in organization-owned groups should be accessible at /organizations/:org/groups/:group_id/assignments/... (view, create, edit, delete, reopen, close, start).
- Standalone groups (not part of any organization) should continue to use their existing flat assignment URLs, unchanged.
- Organization admins and assigned mentors should be able to manage assignments in any group within their organization, not just groups where they are the primary mentor.
- After creating, editing, or deleting an assignment in an organization group, the user should be redirected back to the organization-scoped group page.
- Assignment links throughout the group and assignment views should point to the correct scoped URL based on whether the group belongs to an organization.

### Describe alternatives you've considered

N/A

### Additional context

Project 6 Enterprise & Institutional Organization Features

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by naman79820**
Fixes #7721

 

#### Describe the changes you have made in this PR -
This PR makes assignments work under organization-scoped group URLs, matching how organization-owned groups already behave.

Core changes:

- Assignments in organization-owned groups are now accessible at /organizations/:org/groups/:group_id/assignments/... for all actions (view, create, edit, delete, reopen, close, start). Standalone groups keep their existing flat assignment URLs unchanged.
- Organization admins and assigned mentors can now manage assignments in any group within their organization, not only groups where they are the primary mentor.
- After creating, editing, or deleting an assignment in an organization group, the user is redirected back to the organization-scoped group page.
- Assignment links across the group and assignment views now resolve to the correct scoped URL depending on whether the group belongs to an organization, using a shared helper.

Security and correctness hardening (from review):

- Tenant isolation: Assignment and group lookups are scoped through the organization in the URL, so a request can't reach another organization's group or assignment by manipulating the URL (mismatched IDs return 404).
- Authorization on the new-group form: The new action is now gated the same way as create, so unauthorized users can't view the organization-scoped creation form.
- Feature-flag consistency: The group invite action is gated behind the :organizations flag for organization-scoped requests, keeping organization behavior dormant when the flag is off while leaving standalone invites unaffected.
- Organization-scoped group deletion: Added the organization-scoped destroy route and updated the dashboard delete flow to use it, so deleting a group from the organization dashboard stays organization-scoped.
- Sole-admin protection: An organization must always keep at least one admin. The sole admin can no longer be demoted or removed by anyone (including platform admins). These membership mutations now run inside a row-level lock and reload the record inside the lock, so concurrent requests can't race past the check and leave the organization without an admin.

Cleanup:

- Removed duplicated authorization conditions by delegating to admin_access? where the logic was identical.
- Fixed the assignment mailers to use the assignment's group association and check the flag for the email recipient.
- 
- Tests: Added policy specs (organization admin/mentor/member access to organization-group assignments), request specs (organization-scoped routes, redirects, mismatched-ID isolation), and a sole-admin protection regression test.

### Screenshots of the UI changes (If any) -

---

## Code Understanding and AI Usage

**Did you use AI generated code (ChatGPT, Claude, Copilot, etc.) in any part of this PR? **
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**
Problem: Organization-owned groups use organization-scoped URLs, but their assignments still used flat routes, losing the organization context. Also, only a group's primary mentor could manage its assignments, so organization admins couldn't manage assignments other mentors created.

Approach: I nested assignments (plus reopen/close/start) under the organization-scoped groups so they get organization-scoped URLs, while standalone groups keep their flat URLs. I made AssignmentPolicy organization-aware with a private org_manage? method that returns false for non-organization groups and otherwise delegates to OrganizationGroupPolicy#manage?, so anyone who can manage an organization group can manage its assignments. Controller redirects and view links use shared helpers in the OrganizationScopedRedirect concern that pick the scoped or flat path based on the group. Mailers do the same but check the flag globally since there's no current user.

Alternative considered: Leaving assignments on flat routes, but that keeps URLs inconsistent and doesn't fix the authorization gap, so I scoped them to match the groups.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

* **New Features**
  * Added an organization dashboard with Overview, Members, and Settings tabs.
  * Added branding, social links, group cards, pagination, responsive layouts, and slug availability checking.
  * Added organization-scoped group, assignment, invitation, and member management routes.

* **Bug Fixes**
  * Improved access controls for organization groups, assignments, members, and administrators.
  * Updated notifications, emails, and navigation to use organization-specific links where applicable.
  * Improved organization link validation and protected essential administrators and mentors from removal.

### CODE REVIEW

File: app/policies/organization_member_policy.rb [hunk 1]
```diff
@@ -2,20 +2,21 @@
...
     true
   end
 
   def destroy?
     return false if leaving_self? && sole_admin?
+    return false if member_primary_mentor_of_any_group?
 
-    org_admin? || leaving_self? || user.admin?
+    org_admin? || leaving_self? || user&.admin?
   end
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/policies/organization_member_policy.rb:20]: check it should be resolved @coderabbitai

File: app/views/organizations/overview.html.erb [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                </div>
+                <div class="d-flex justify-content-between align-items-end gap-2 mt-3">
+                  <span class="org-dash-group-meta d-inline-flex align-items-center gap-1">
+                    <i class="fa fa-user" aria-hidden="true"></i>
+                    <%= t(".member_count", count: group.group_members_count) %>
+                  </span>
+                  <% if OrganizationGroupPolicy.new(current_user, group).manage? %>
+                    <%= link_to "#", class: "org-dash-group-icon org-dash-group-icon-delete",
+      data: { bs_toggle: "modal", bs_target: "#deletegroupModal", currentgroup: group.id },
+      aria: { label: t("delete") } do %>
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/views/organizations/overview.html.erb:69]: check it should be resolved @coderabbitai

**Comment by naman79820** [comment]
too much conflicts. Creating new pr.
